### ISSUE TITLE
local install freezes thunderbird and virtual server install listens on 127.0.0.1?

### ISSUE THREAD

**Issue by gstlouisgit** [nuald/thb-gramchecker#21]
I have thunderbird 68.6.0 (64-bit)
I installed Grammar Checker from the extensions in thunderbird.

For some reason I could get grammar to work out of the box I could see it was searching for https://languagetool.org/api/ which failed, then I found I could change this going TBpreferences->advanced->Config Editor, search by grammar and find extensions.grammarchecker.urlpref1; and set value to https://languagetool.org/api/v2/check
This would fail as well saying "Cannot connect to the LanguageTool server!"

Then I followed the instructions on the main page about doing local install

```
$ wget https://languagetool.org/download/LanguageTool-4.6.zip
$ unzip LanguageTool-4.6.zip
$ cd LanguageTool-4.6
$ java -cp languagetool-server.jar org.languagetool.server.HTTPServer --port 8081
```

I changed the url to http://localhost:8081 and it connected, took a very long time, I see in the terminal it did its check, then thunderbird just freezes completely freezes.  needs a force quit.

I als tried the same setup on a local virtual machine and the problem with this is that it starts up with `java -cp languagetool-server.jar org.languagetool.server.HTTPServer `
but then 8081 port isn't open, no firewall and doing netstat -tlpn | grep 8081 gives me 
`tcp6       0      0 127.0.0.1:8081          :::*                    LISTEN      10188/java `
its like its listening only on loopback?

I feel I'm doing a lot of things wrong here...

**Comment by gstlouisgit** [comment]
Some feedback on this problem and how it was solved.

**I believe the reason this was slow ** is that when it was checking the text in the email it was literally going through the whole thread of the email... so replies already done etc. it was reading  everything!

**I wasn't connecting to public because** I did not use the --public flag.  For instance:
```
java -cp languagetool-server.jar org.languagetool.server.HTTPServer --port 808 --public 1 --allow-origin "*" &
```
The ampersand at the end is because I executed this in the background so that it is always running.  I thought doing it this way, if I logged out it would terminate but from what I can see it is still running.

If anyone has a better way to run this as a type of daemon service so it loads on boot pleas share.

**Comment by nuald** [comment]
Please note that the `public` option is not about connecting to some public space, but rather the flag to allow the server to accept the incoming connections from any client, not the local ones only: 
```
$ java -cp languagetool-server.jar org.languagetool.server.HTTPServer --help
...
  --public         allow this server process to be connected from anywhere; if not set,
                   it can only be connected from the computer it was started on
```
I guess if you use VM, you have to use that flag, therefore the background start command would be the same as you've described:
```
java -cp languagetool-server.jar org.languagetool.server.HTTPServer --port 8081 --public --allow-origin "*" &
```
As for the daemon mode, I guess I can help with the script, but it depends on the OS (in particular, the init daemon - I guess the most common is `systemd` nowadays, but you may have the different one or even not using GNU/Linux at all). Please let me know the OS (and if it's GNU/Linux, the exact distribution and its number), and I'll try to help.

**Comment by gstlouisgit** [comment]
Thanks for the reply.
I will be using CentOS 7.

For Thunderbird plugin.  
I have found a few problems.  

- when checking text on a reply email, it runs through all the complete email text.  It then expands the email reply window vertically full size where you cannot scroll, cannot shrink the window vertically and I cannot see it offering any correction because I cannot see the bottom of the email message ( this seems to only happen to reply emails )

- when doing a check on a simple text, it does respond and provides feedback at the bottom, but it doesn't really offer simple suggestions about error texts.  I have attached an img

- When java server is on with --public and --allow-origin "*" flags I use http://server:808/v2/ which works in chrome plugin.  However in TB only http://server:808 works.  I put this in TB->preferences->Config Editor->search grammar and add url in 'extensions.grammarchecker.urlpref1'.  So I can see the reply suggestions at the bottom of the email but it doesn't really seem to offer suggestions on my errors - so kind of doesn't work.

<img width="1035" alt="Screen Shot 2020-04-04 at 11 05 06 AM" src="https://user-images.githubusercontent.com/20375753/78454343-84340e80-7665-11ea-9e49-05aadc3013cc.png">
